File: solutions/js/homework-nft.js
// ============================================================
// Homework Solution: NFT Lifecycle
// Learn: How to mint, query, and burn NFTs on the XRP Ledger.
// You'll experience the full lifecycle of an NFToken.
// ============================================================

const xrpl = require('xrpl');
const fs = require('fs');

const TESTNET_URL = 'wss://s.altnet.rippletest.net:51233';

async function main() {
  const { wallets } = JSON.parse(fs.readFileSync('../../wallets.json', 'utf-8'));
  const wallet = xrpl.Wallet.fromSeed(wallets[0].seed);

  const client = new xrpl.Client(TESTNET_URL);
  await client.connect();
  console.log('Connected to XRPL Testnet');

  // Mint NFT
  const mintTx = {
    TransactionType: 'NFTokenMint',
    Account: wallet.address,
    NFTokenTaxon: 0,
    Flags: 8,
    TransferFee: 5000,
    URI: xrpl.convertStringToHex('ipfs://YOUR_NAME_HERE')
  };
  const mintResult = await client.submitAndWait(mintTx, { wallet });
  console.log('NFT Minted!', mintResult.result.hash);

  // Query NFTs
  const nfts = await client.request({
    command: 'account_nfts',
    account: wallet.address
  });
  const tokenId = nfts.result.account_nfts[0].NFTokenID;
  console.log('NFTokenID:', tokenId);

  // Burn NFT
  const burnTx = {
    TransactionType: 'NFTokenBurn',
    Account: wallet.address,
    NFTokenID: tokenId
  };
  const burnResult = await client.submitAndWait(burnTx, { wallet });
  console.log('NFT Burned!', burnResult.result.hash);

  // Verify it's gone
  const nftsAfter = await client.request({
    command: 'account_nfts',
    account: wallet.address
  });
  console.log('NFTs remaining:', nftsAfter.result.account_nfts.length);

  console.log(`\nExplorer: https://testnet.xrpl.org/accounts/${wallet.address}`);
  await client.disconnect();
}

main().catch(console.error);

File: solutions/python/homework_nft.py
"""
Homework Solution: NFT Mint, Query, and Burn
=============================================
Mint an NFT, query it, then burn it on XRPL Testnet.
"""

import json
from xrpl.clients import WebsocketClient
from xrpl.wallet import Wallet
from xrpl.models.transactions import NFTokenMint, NFTokenBurn
from xrpl.models.requests import AccountNFTs
from xrpl.transaction import submit_and_wait
from xrpl.utils import str_to_hex

TESTNET_URL = 'wss://s.altnet.rippletest.net:51233'

# Load wallets
with open('../../wallets.json') as f:
    data = json.load(f)
    wallets = data['wallets']

wallet = Wallet.from_seed(wallets[0]['seed'])
print(f"Wallet: {wallet.address}")

with WebsocketClient(TESTNET_URL) as client:

    # Mint NFT
    mint_tx = NFTokenMint(
        account=wallet.address,
        nftoken_taxon=0,
        flags=8,
        transfer_fee=5000,
        uri=str_to_hex('ipfs://YOUR_NAME_HERE')
    )
    mint_result = submit_and_wait(mint_tx, client, wallet)
    print(f"Mint result: {mint_result.result['meta']['TransactionResult']}")

    # Query NFTs
    nfts_response = client.request(AccountNFTs(account=wallet.address))
    nfts = nfts_response.result['account_nfts']
    print(f"NFTs owned: {len(nfts)}")
    nftoken_id = nfts[-1]['NFTokenID']
    print(f"NFToken ID: {nftoken_id}")

    # Burn NFT
    burn_tx = NFTokenBurn(
        account=wallet.address,
        nftoken_id=nftoken_id
    )
    burn_result = submit_and_wait(burn_tx, client, wallet)
    print(f"Burn result: {burn_result.result['meta']['TransactionResult']}")

    # Verify burned
    nfts_response = client.request(AccountNFTs(account=wallet.address))
    nfts = nfts_response.result['account_nfts']
    print(f"NFTs owned after burn: {len(nfts)}")

# Explorer: https://testnet.xrpl.org/accounts/{address}/nfts
print(f"\nExplorer: https://testnet.xrpl.org/accounts/{wallet.address}/nfts")
